server/elasticsearch/src/main/java/org/hawkular/apm/server/elasticsearch/SpanServiceElasticsearch.java

```diff
@@ -0,0 +1,196 @@
+
+            client.getClient()
+                    .admin()
+                    .indices()
+                    .refresh(refreshRequestBuilder.request())
+                    .actionGet();
+
+            QueryBuilder query = QueryBuilders.termQuery("parentId", id);
+
+            SearchRequestBuilder request = client.getClient()
+                    .prepareSearch(index)
+                    .setTypes(SPAN_TYPE)
+                    .setSearchType(SearchType.DFS_QUERY_THEN_FETCH)
+                    .setQuery(query);
+
+            SearchResponse response = request.execute()
+                    .actionGet();
+
+            for (SearchHit searchHitFields : response.getHits()) {
+                try {
+                    spans.add(deserialize(searchHitFields.getSourceAsString(), Span.class));
+                } catch (IOException ex) {
+                    msgLog.errorFailedToParse(ex);
+                }
+            }
+        } catch (IndexMissingException ex) {
+            msgLog.tracef("No index found, so unable to retrieve spans");
+        }
+
+        msgLog.tracef("Get children with parentId[%s] is: %s", id, spans);
+        return spans;
+    }
+
+    @Override
+    public void storeSpan(String tenantId, List<Span> spans) throws StoreException {
+        storeSpan(tenantId, spans, span -> span.getId());
+    }
+
+    @Override
+    public void storeSpan(String tenantId, List<Span> spans, Function<Span, String> spanIdSupplier)
+            throws StoreException {
+        client.initTenant(tenantId);
+
+        BulkRequestBuilder bulkRequestBuilder = client.getClient().prepareBulk();
+
+        try {
+            for (Span span: spans) {
+                // modified id is used in index
+                final String modifiedId = spanIdSupplier.apply(span);
+                String json = serialize(span);
+
+                msgLog.tracef("Storing span: %s", json);
+
+                bulkRequestBuilder
+                        .add(client.getClient()
+                                .prepareIndex(client.getIndex(tenantId), SPAN_TYPE, modifiedId)
+                                .setSource(json));
+            }
+        } catch (IOException ex) {
+            throw new StoreException(ex);
+        }
+
+        BulkResponse bulkItemResponses = bulkRequestBuilder.execute().actionGet();
+
+        if (bulkItemResponses.hasFailures()) {
+            msgLog.tracef("Failed to store traces to elasticsearch: %s", bulkItemResponses.buildFailureMessage());
+            throw new StoreException(bulkItemResponses.buildFailureMessage());
+        }
+
+        msgLog.trace("Success storing traces to elasticsearch");
+
+    }
+
+    @Override
+    public void clear(String tenantId) {
+        String index = client.getIndex(tenantId);
+
+        try {
+            client.getClient().admin().indices().prepareDelete(index).execute().actionGet();
+            client.clear(tenantId);
+        } catch (IndexMissingException ime) {
+            // Ignore
```

Comment:
Is there an operation where you can check if the index exists? Exceptions are expensive, so, if there's a "good" chance that an index doesn't exist, it's better to check it first.